### PR TITLE
Feature/ab2d-3990 Record Number of Patients With Eobs

### PR DESCRIPTION
**Pull request by sb-wnyffenegger**
**JIRA Tickets:**

[AB2D-3990](https://jira.cms.gov/browse/AB2D-3990) - Record Number of Patients With Eobs
 
### What Does This PR Do?

Create a statistic regarding the number of patients who, after filtering, have at least one EOB associated with them in every job.

Changes:

1. Create column in event_bene_search for this table and add liquibase change to database changelog.
2. Update ContractSearchEventMapper and associated tests to check that this field is successfully serialized and deserialized.
3. Create new field in ProgressTracker and ProgressTrackerUpdate to track the statistic.
4. Increment the ProgressTrackerUpdate instance whenever a patient with eobs is found.
4. Create a new JobMeasure enum value that allows updating this statistic for any given running job.
5. Update integration test with the value of this statistic.

### What Should Reviewers Watch For?

Any potential miscalculations of the statistic.

### Security Implications

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [ ] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [ ] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [ ] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [ ] Code checked for PHI/PII exposure

### CODE REVIEW

File: eventlogger/src/test/java/gov/cms/ab2d/eventlogger/reports/sql/LoggerEventSummaryTest.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ void getSummaryBasic() {
         LoggableEvent e4 = new FileEvent(usr, jobId, file1, FileEvent.FileStatus.OPEN);
         e4.setTimeOfEvent(firstTime.plusDays(3));
         logger.log(e4);
-        LoggableEvent e5 = new ContractSearchEvent(usr, jobId, "CONTRACT1", 100, 100, 100, 2, 1000, 1000, 1);
+        LoggableEvent e5 = new ContractSearchEvent(usr, jobId, "CONTRACT1", 100, 100, 100, 2, 100, 1000, 1000, 1);
```
Comment (nflearl) [inline eventlogger/src/test/java/gov/cms/ab2d/eventlogger/reports/sql/LoggerEventSummaryTest.java:54]: Reusing the value of 100 open up a cross-wire vulnerabilty i.e. if one has two fields a and b and sets them both to the same value, then one can not detect if b was updated when a should be.  

Consider using unique values for each member field.

File: worker/src/test/java/gov/cms/ab2d/worker/processor/JobProcessorIntegrationTest.java [hunk 2]
```diff
@@ -254,6 +254,7 @@ void when_errorCount_is_below_threshold_do_not_fail_job() {
         assertEquals(100, event.getBenesExpected());
         assertEquals(CONTRACT_NAME, event.getContractNumber());
         assertEquals(100, event.getBenesSearched());
+        assertEquals(100, event.getBenesWithEobs());
```
Comment (nflearl) [inline worker/src/test/java/gov/cms/ab2d/worker/processor/JobProcessorIntegrationTest.java:257]: Reusing the value of 100 open up a cross-wire vulnerabilty i.e. if one has two fields a and b and sets them both to the same value, then one can not detect if b was updated when a should be.  

Consider using unique values for each member field.

**Review by nflearl — APPROVED** [review]

File: eventlogger/src/test/java/gov/cms/ab2d/eventlogger/reports/sql/LoggerEventSummaryTest.java (hunk 1 above)
Comment (sb-dkniesteadt) [reply eventlogger/src/test/java/gov/cms/ab2d/eventlogger/reports/sql/LoggerEventSummaryTest.java:54]: agread

File: worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java [hunk 3]
```diff
@@ -399,6 +400,8 @@ private void updateJobProgress(ContractData contractData, ProgressTrackerUpdate
         jobChannelService.sendUpdate(jobUuid, JobMeasure.PATIENT_REQUESTS_ERRORED,
                 updateTracker.getPatientFailureCount());
 
+        jobChannelService.sendUpdate(jobUuid, JobMeasure.PATIENTS_WITH_EOBS, updateTracker.getPatientWithEobCount());
```
Comment (sb-dkniesteadt) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java:403]: small, Have updateTracker.getPatientWithEobCount()); on the next line like the rest.

**Review by sb-dkniesteadt — APPROVED** [review]

**Comment by sb-wnyffenegger** [comment]
@nflearl @sb-dkniesteadt addressed PR comments

**Review by nflearl — APPROVED** [review]

**Review by sb-dkniesteadt — APPROVED** [review]

**Pull request merged**